spec/handlers/newflow/confirm_oauth_info_spec.rb

```diff
@@ -24,37 +21,91 @@ module Newflow
       }
     end
 
-    it 'adds the user as a "lead" to salesforce' do
-      expect_any_instance_of(PushSalesforceLead).to receive(:exec)
-      described_class.call(params: params, user: User.last)
-    end
+    context 'when user has an email address' do
+      before do
+        user = FactoryBot.create(:user, state: 'unverified')
+        FactoryBot.create(:email_address, value: params[:signup][:email], user: user)
+      end
 
-    it 'changes the user state to "activated"' do
-      expect(User.last.state).to_not eq('activated')
-      described_class.call(params: params, user: User.last)
-      expect(User.last.state).to eq('activated')
-    end
+      it 'adds the user as a "lead" to salesforce' do
+        expect_any_instance_of(PushSalesforceLead).to receive(:exec)
+        described_class.call(params: params, user: User.last)
+      end
+
+      it 'changes the user state to "activated"' do
+        expect(User.last.state).to_not eq('activated')
+        described_class.call(params: params, user: User.last)
+        expect(User.last.state).to eq('activated')
+      end
 
-    it 'agrees to terms of use and privacy policy when contracts_required' do
-      # TODO: ideally would do this but it fails with an error:
-      # expect_any_instance_of(AgreeToTerms).to receive(:call).twice.and_call_original
+      it 'agrees to terms of use and privacy policy when contracts_required' do
+        # TODO: ideally would do this but it fails with an error:
+        # expect_any_instance_of(AgreeToTerms).to receive(:call).twice.and_call_original
 
-      expect {
+        expect {
+          described_class.call(params: params, contracts_required: true, user: User.last)
+        }.to change {
+          FinePrint::Signature.count
+        }.by(2)
+      end
+
+      it 'signs up user for the newsletter when checked' do
+        expect_any_instance_of(PushSalesforceLead).to receive(:exec).with(hash_including({ newsletter: true }))
         described_class.call(params: params, contracts_required: true, user: User.last)
-      }.to change {
-        FinePrint::Signature.count
-      }.by(2)
-    end
+      end
 
-    it 'signs up user for the newsletter when checked' do
-      expect_any_instance_of(PushSalesforceLead).to receive(:exec).with(hash_including({ newsletter: true }))
-      described_class.call(params: params, contracts_required: true, user: User.last)
+      it 'does NOT sign up user for the newsletter when NOT checked' do
```

Comment:
this would read better if the NOT checked is set in a context.  it tests generally should avoid compound statements like this  it does X when Y and Z sort of thing.  Y and Z are usually set as context blocks.